Cubical/Homotopy/Freudenthal.agda

```diff
@@ -0,0 +1,131 @@
+{-
+
+Freudenthal suspension theorem
+
+-}
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Homotopy.Freudenthal where
+
+open import Cubical.Foundations.Everything
+open import Cubical.Data.HomotopyGroup
+open import Cubical.Data.Nat
+import Cubical.Data.NatMinusTwo as ℕ₋₂
+open import Cubical.Data.Sigma
+open import Cubical.HITs.Nullification
+open import Cubical.HITs.Susp
+open import Cubical.HITs.Truncation as Trunc
+open import Cubical.Homotopy.Connected
+open import Cubical.Homotopy.WedgeConnectivity
+
+private
+  -- This alternative definition of rCancel will make the proof smoother
+
+  rCancel-filler' : ∀ {ℓ} {A : Type ℓ} {x y : A} (p : x ≡ y) → (i j k : I) → A
+  rCancel-filler' {x = x} {y} p i j k =
+    hfill
+      (λ i → λ
+        { (j = i1) → p (~ i ∧ k)
+        ; (k = i0) → x
+        ; (k = i1) → p (~ i)
+        })
+      (inS (p k))
+      (~ i)
+
+  rCancel' : ∀ {ℓ} {A : Type ℓ} {x y : A} (p : x ≡ y) → p ∙ p ⁻¹ ≡ refl
+  rCancel' p j k = rCancel-filler' p i0 j k
+
+module _ {ℓ} (n : ℕ)
+  {A : Pointed ℓ} (connA : isHLevelConnected (suc (suc n)) (typ A))
+  where
+
+  σ : typ A → typ (Ω (∙Susp (typ A)))
+  σ a = merid a ∙ merid (pt A) ⁻¹
+
+  private
+    2n+2 = suc n + suc n
+
+    module WC (p : north ≡ north) =
+      WedgeConnectivity (suc n) (suc n) A connA A connA
+        (λ a b →
+          ( (σ b ≡ p → hLevelTrunc 2n+2 (fiber (λ x → merid x ∙ merid a ⁻¹) p))
+          , isOfHLevelPi 2n+2 λ _ → isOfHLevelTrunc 2n+2
+          ))
+        (λ a r → ∣ a , (rCancel' (merid a) ∙ rCancel' (merid (pt A)) ⁻¹) ∙ r ∣)
+        (λ b r → ∣ b , r ∣)
+        (funExt λ r →
+          cong ∣_∣
+            (cong (pt A ,_)
+              (cong (_∙ r) (rCancel' (rCancel' (merid (pt A)))) ∙ lUnit r ⁻¹)))
```

Comment:
Is there some reason to do `cong` 3 times in a row instead of just once with the composite of the functions?

Reply:
I think I was getting yellow otherwise

Reply:
Maybe cong' would have helped: https://github.com/agda/cubical/blob/master/Cubical/Foundations/Path.agda#L16

Reply:
That worked to combine the outer two. The inner one is still separate because the shape is cong (...) (cong (...) (...) ∙ lUnit r ⁻¹).

Reply:
Cool!